core/src/transport.rs

```diff
@@ -130,6 +130,19 @@ pub trait Transport {
     where
         Self: Sized;
 
+    /// As [`Transport::dial`] but overrides the role of the local node on the connection . I.e. has
+    /// the local node act as a listener on the outgoing connection.
+    ///
+    /// This option is needed for NAT and firewall hole punching.
+    ///
+    /// See [`ConnectedPoint::Dialer`](crate::connection::ConnectedPoint::Dialer) for related option.
+    fn dial_with_role_override(
```

Comment:
I think on this level, it might be more descriptive to use a terminology like `dial_as_listener`? There is no overriding parameter here so not having been involved this discussion, I would be thinking "overridden with what?"